src/R/Editor/Impl/Completion/Providers/PackageFunctionCompletionProvider.cs

```diff
@@ -153,19 +156,23 @@ public class PackageFunctionCompletionProvider : IRCompletionListProvider, IRHel
         /// </summary>
         /// <param name="context"></param>
         /// <returns></returns>
-        private IEnumerable<IPackageInfo> GetAllFilePackages(RCompletionContext context) {
+        private async Task<IEnumerable<IPackageInfo>> GetAllFilePackagesAsync(RCompletionContext context) {
             _loadedPackagesProvider?.Initialize();
 
             IEnumerable<string> loadedPackages = _loadedPackagesProvider?.GetPackageNames() ?? Enumerable.Empty<string>();
             IEnumerable<string> filePackageNames = context.AstRoot.GetFilePackageNames();
             IEnumerable<string> allPackageNames = PackageIndex.PreloadedPackages.Union(filePackageNames).Union(loadedPackages);
 
-            return allPackageNames
-                .Select(packageName => GetPackageByName(packageName))
+            var list = new List<IPackageInfo>();
+            foreach (var name in allPackageNames) {
+                var info = await _packageIndex.GetPackageInfoAsync(name);
```

Comment:
Does `GetPackageInfoAsync` do any R evals or anything else that requires a roundtrip to the host? If so, we might want to make it a batch API, so that it can do it for all these packages in one go, to avoid unnecessary latency on every request.